pkg/evs/controllerserver.go

```diff
@@ -264,20 +267,189 @@ func (cs *ControllerServer) ListVolumes(_ context.Context, req *csi.ListVolumesR
 	return nil, status.Error(codes.Unimplemented, "")
 }
 
-func (cs *ControllerServer) CreateSnapshot(_ context.Context, req *csi.CreateSnapshotRequest) (
-	*csi.CreateSnapshotResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "")
+func (cs *ControllerServer) CreateSnapshot(_ context.Context, req *csi.CreateSnapshotRequest) (*csi.CreateSnapshotResponse, error) {
+	klog.V(4).Infof("CreateSnapshot called with request %v", *req)
+	credentials := cs.Driver.cloudCredentials
+	name := req.GetName()
+	volumeId := req.GetSourceVolumeId()
+	if err := checkCreateSnapshotParamEmpty(name, volumeId); err != nil {
+		return nil, err
+	}
+	if response, err := checkRepeatSnapshotName(credentials, name, volumeId); response != nil || err != nil {
```

Comment:
suggest to name the function to checkDuplicateSnapshotName instead of checkRepeatSnapshotName.